File: Python/8kyu/8kyu-square(n)-sum.py
# Square(n) Sum
# Difficulty: 8 kyu

def square_sum(numbers):
    return sum([n ** 2 for n in numbers])

# OR

def square_sum(numbers):
    return sum(n ** 2 for n in numbers)


# Tests
import codewars_test as test
from solution import square_sum

@test.describe("Fixed Tests")
def basic_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        test.assert_equals(square_sum([1,2]), 5)
        test.assert_equals(square_sum([0, 3, 4, 5]), 50)
        test.assert_equals(square_sum([]), 0)
        test.assert_equals(square_sum([-1,-2]), 5)
        test.assert_equals(square_sum([-1,0,1]), 2)

File: R/8kyu/8kyu-square(n)-sum.r
# Square(n) Sum
# Difficulty: 8 kyu

square_sum <- function(nums) {
  sum(nums^2)
}


# Tests

test_that("basic tests", {
  expect_equal(square_sum(c(1, 2)), 5)
  expect_equal(square_sum(c(0, 3, 4, 5)), 50)
  expect_equal(square_sum(c()), 0)
  expect_equal(square_sum(c(-1, -2)), 5)
  expect_equal(square_sum(c(-1, 0, 1)), 2)
})